src/pages/PublishedSchedule/PublishedSchedule.jsx

```diff
@@ -31,32 +33,38 @@ const PublishedSchedule = () => {
 
   const curSeason = getTodaySeason();
 
-  useEffect(() => {
-    const renderTable = async () => {
-      const { data } = await NPOBackend.get('/published-schedule/all-seasons');
-      if (data.indexOf(curSeason) == -1) {
-        data.unshift(curSeason);
-      }
+  const renderSeasons = async () => {
+    const { data } = await NPOBackend.get('/published-schedule/all-seasons');
+    if (data.indexOf(curSeason) == -1) {
+      data.unshift(curSeason);
+    }
 
-      if (selectedSeason === '') {
-        setSelectedSeason(curSeason);
+    const seasonOrder = ['Fall', 'Summer', 'Spring'];
+    data.sort((a, b) => {
+      // Compare years first
+      if (a.split(' ')[1] !== b.split(' ')[1]) {
+        return b.split(' ')[1] - a.split(' ')[1];
+      } else {
+        return seasonOrder.indexOf(a.split(' ')[0]) - seasonOrder.indexOf(b.split(' ')[0]);
       }
+    });
 
-      const seasonOrder = ['Fall', 'Summer', 'Spring'];
-      data.sort((a, b) => {
-        // Compare years first
-        if (a.split(' ')[1] !== b.split(' ')[1]) {
-          return b.split(' ')[1] - a.split(' ')[1];
-        } else {
-          return seasonOrder.indexOf(a.split(' ')[0]) - seasonOrder.indexOf(b.split(' ')[0]);
-        }
-      });
+    setAllSeasons(data);
+  };
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>The 'renderSeasons' function makes the dependencies of useEffect Hook (at line 67) change on every render. To fix this, wrap the definition of 'renderSeasons' in its own useCallback() Hook.